TypeCobol.LanguageServer/VsCodeProtocol/Telemetry Notification/TelemetryEvent.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Text;
+
+namespace TypeCobol.LanguageServer.VsCodeProtocol
+{
+    /// <summary>
+    /// Class used to describe an event that occured on this server.
+    /// </summary>
+    public class TelemetryEvent
+    {
+        /// <summary>
+        /// Extract data from an exception to produce a meaningful TelemetryEvent.
+        /// The event will contain :
+        /// - the exception's .NET type
+        /// - the exception's message
+        /// - the exception's target site (which is the method that threw the exception, top of the StackTrace)
+        /// </summary>
+        /// <param name="exception">Non-null exception instance to describe.</param>
+        /// <returns>Non-null instance of TelemetryEvent.</returns>
+        public static TelemetryEvent CreateFrom(Exception exception)
+        {
+            string exceptionType = exception.GetType().FullName;
+            string exceptionMessage = exception.Message;
+            var exceptionTargetSite = new StringBuilder();
+            if (exception.TargetSite != null)
+            {
+                var targetSite = exception.TargetSite;
+                exceptionTargetSite.Append(targetSite.DeclaringType?.FullName); // Fullname of the type declaring the method
+                exceptionTargetSite.Append('.');
+                exceptionTargetSite.Append(targetSite.Name);
+                exceptionTargetSite.Append('(');
+                exceptionTargetSite.AppendJoin(", ", targetSite.GetParameters().Select(p => p.ParameterType.Name)); // For each param, simple name of its type
+                exceptionTargetSite.Append(')');
+            }
```

Comment:
Maybe the `targetSite.Module` could be interesting as well?

Reply:
Yes this will help us group exceptions. However I prefer using the `Exception.Source` property instead of adding somtheing in the `TargetSite` string description